docs/proposals/20230424-unify-the-definition-of-functions.md

```diff
@@ -0,0 +1,123 @@
+# Unify the definition of sync and async functions
+
+## Motivation
+Currently, we use `runtime: knative` and `runtime: async` to distinguish sync and async functions which increases the learning curve. Actually the difference between sync and async functions is trigger type:
+- Sync functions are triggered by `HTTP` event, which are defined by specifying `runtime: knative`
+- Async functions are triggered by events from components of `Dapr bindings` or `Dapr pubsub`. `runtime: async` and `inputs` have to be used together to specify triggers for async functions
+
+So we can use `triggers` to replace `runtime` and `inputs`.
+
+## Goals
+Use `triggers` to unify the definition of sync and async functions.
+
+## Proposal
+
+### OpenFunction
+
+We can use `triggers` to replace the original `inputs` field. The original `runtime` field will be replaced by `trigger.type`
+The API version can be upgrade from v1beta1 to v1
```

Comment:
I feel like we should keep it in `v1beta1` and promote to `v1` once it is stable enough.

Reply:
> I feel like we should keep it in `v1beta1` and promote to `v1` once it is stable enough.

There has to be a API conversion between this proposal and the old one, so we should bump the API version.

Reply:
I think so too.

Reply:
how about `v1beta2`? usually `v1` means it reaches [GA/stable](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-changes) state.